### PR TITLE
Begin documenting supported (target) platforms

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

**Comment by ogoffart** [comment]
A few comments:
 - I wonder if it is worth trying to explicitly name all these versions, just name the latest we support.  This is what is usually done.  
 - We could also do it without naming anything, but just stating the rule of what we're supporting.
 - Especially for linux this is a bit strange to name Ubuntu only. While it's true that it's what is build in the CI, the CI don't test what's most important (running the app and making sure it actually display stuff)
 - As I said, i'm not a fan of duplicated contents like this in all our "book".  I think we need to restructure the documentation in https://slint.dev/releases/1.1.1/docs/slint/ to put the "slint language documentation" one level bellow, so we can have some other chapters to document general things such as the supported platforms, the documentation for the slint-viewer, the documentation for slintpad, the documentation of the environment variable for debugging, and so on.  We discussed the one book vs many book in https://github.com/slint-ui/slint/issues/793#issuecomment-1423982360  and i don't remember which of these option we decide on. But i think we don't want duplicated contents.)
 - Regarding the language specific, this might also be the place to document the minimum compiler versions and such (but yeah, this can be done later and not in this PR)
 - Should we specify things like deployment vs host platform?
 - Supported browser for our wasm build / slintpad?

So now the question is really what we want to support.
I think what you wrote is fine.
> The general objective is to support the operating systems that are supported by their vendors by the time Slint is released.

But really what do we need to "support" a platform, is that it is supported by our upstream:
 - rust:  https://doc.rust-lang.org/nightly/rustc/platform-support.html
 - winit: https://github.com/rust-windowing/winit/blob/master/FEATURES.md#winit-scope
 - femtovg: ?
 - skia: https://skia.org/about/  https://skia.org/docs/user/build/#supported-and-preferred-compilers
 - Qt (for the qt backend) https://doc.qt.io/qt-5/supported-platforms.html / https://doc.qt.io/qt-6/supported-platforms.html
 
Maybe it's actually enough to just link to these resources.

We can have a more strict rule saying something like
- What we test regularily are: eg: the latest version of each operating system.
- What we maintain working (fix bugs if reported on our bug tracker):  eg: what is supported by the OS vendors.  But that said, the very old redhat/centos that are still supported can be challenging. espcially as build platform.
- Everything else needs a commercial support contract.

**Comment by tronical** [comment]
I've made some changes now.

* I've removed the version. Yes, that doesn't really have to be there, it belongs into a more prominent spot -> https://github.com/slint-ui/slint/pull/3277 .
* The document is now centrally in the Slint reference, no duplication anymore.
* I've tried to emphasise in the wording that this is about the target deployment platforms. I'd leave the development environment for another section.
* It's true that we could link to the rust platform and winit scope docs, but... then those need to be also matched against whatever rust version was current at that point, i.e. the links need to be versioned?

 I'm also not entirely sure what the correct format here is though - listing an "random" Ubuntu version seems odd. At the same time... when did a specific "Linux distro" cause any issues that would be worth mentioning here?

So another option would be to do this more "demand driven", i.e. we build this page over time based on what we see is necessary. Right now we know that the Windows version and macOS version is something we've needed to document, so I think we should keep that there (and update before each release as needed).

For Linux, we could replace the table with a generic paragraph that says that it should run on the most recent LTS or newer release of distros. After all, there's not much development / change going on on the Linux desktop anymore. I've done that in the latest revision.

Then, in the future, when we run into specific situations we can extend this documentation with quirks or tables.

File: docs/reference/src/introduction/supported_platforms.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Generally, Slint runs on Windows, macOS, and popular Linux distributions. The following tables below
+cover versions that we specifically test. The general objective is to support the operating systems that
+are supported by their vendors by the time a Slint version is released.
+
+### Windows
+
+| Operating System | Architecture |
+|------------------|--------------|
+| Windows 10       | x86-64       |
```
Comment (ogoffart) [inline docs/reference/src/introduction/supported_platforms.md:20]: Do we test Windows 10?  What does that mean that we test it?

**Review by ogoffart — APPROVED** [review]
I wonder how easy is it going to maintain this page? Or if we're going to forgot adding new versions of Os's as they are released, and remove old ones we don't "test"

File: docs/reference/src/introduction/supported_platforms.md (hunk 1 above)
Comment (tronical) [reply docs/reference/src/introduction/supported_platforms.md:20]: We run our tests on Windows Server 2022, which is the Windows 10 code base AFAIK.

**Pull request merged**